pytest_mitmproxy.py

```diff
@@ -1 +1,106 @@
 # -*- coding: utf-8 -*-
+import re
+import os
+import six
+import time
+import shutil
+import pytest
+from contextlib import contextmanager
+from tempfile import TemporaryFile
+if six.PY3:
+    from urllib.parse import urlparse
+    from subprocess import Popen, CalledProcessError, TimeoutExpired
+elif six.PY2:
+    from urlparse import urlparse
+    from subprocess import Popen, CalledProcessError
+
+# clean the dumps each run
+if os.path.isdir("traffic_dumps"):
+    shutil.rmtree("traffic_dumps")
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--mitmproxy-filter",
+        action="store",
+        default=None,
+        help="filter specification for mitmproxy"
+    )
+
+
+@contextmanager
+def new_proxy(name, scope, flt=None):
+    ''' Start a new mitmdump process and
+    return the address for clients to connect
+
+    The captured content will be written to
+    ./<dump_dir>/<scope>/<name>
+    '''
+    dump_dir = "traffic_dumps"
+
+    if six.PY3:
+        os.makedirs(os.path.join(dump_dir, scope), mode=0o755, exist_ok=True)
+    elif six.PY2:
+        try:
+            os.makedirs(os.path.join(dump_dir, scope), mode=0o755)
+        except OSError:
+            pass
```

Comment:
You don't really need two code paths here, but that error checking isn't good enough...

```python
try:
    os.makedirs(os.path.join(dump_dir, scope), mode=0o755)
except OSError as err:
    if err.errno != errno.EEXIST:
        raise
```

Should work correctly on both python2 and python3:

Note that just blindly catching OSError will trigger on lots of other errors, for example, on permission problems, memory or inode exhaustion, path too long, etc.